resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -79,6 +80,19 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
 
   private val yarnConf = new YarnConfiguration(SparkHadoopUtil.newConfiguration(sparkConf))
 
+  if (isClusterMode) {
+    // this logic replicates the way YARN NM determines the address to bind listeners to
```

Comment:
This explain what it does, but it should be explaining why it does it.

Reply:
please review my updated comment